profit/al/active_learning.py

```diff
@@ -1,137 +1,61 @@
 
     Parameters:
         runner (profit.run.Runner): Runner to dynamically start runs.
-        surrogate (profit.sur.Surrogate): Surrogate used for fitting.
-        variables (profit.util.variable_kinds.VariableGroup): Variables.
+        variables (profit.util.variable.VariableGroup): Variables.
         ntrain (int): Total number of training points.
-        nwarm (int): Number of warmup (random) initialization points.
+        nwarmup (int): Number of warmup (random) initialization points.
         batch_size (int): Number of training samples learned in parallel.
-        acquisition_function (str/profit.al.acquisition_functions.AcquisitionFunction): Acquisition function used for
-            selecting the next candidates.
         convergence_criterion (float): AL is stopped when the loss of the acquisition function is lower than this
             criterion. Not implemented yet.
-        nsearch_points (int): Number of possible candidate points in each dimension.
+        nsearch (int): Number of possible candidate points in each dimension.
         make_plot (bool): Flat indicating if the AL progress is plotted.
 
     Attributes:
-        search_space (dict[str, np.array]): np.linspace for each AL input variable.
-        Xpred (np.array): Matrix of the candidate points built with np.meshgrid.
         krun (int): Current training cycle.
     """
+    labels = {}
 
-    _models = {}
-
-    def __init__(self, runner, surrogate, variables, ntrain, nwarm=1, batch_size=1,
-                 acquisition_function='simple_exploration', convergence_criterion=1e-5, nsearch_points=50, make_plot=False):
-        from profit.al.aquisition_functions import AcquisitionFunction
-
+    def __init__(self, runner, variables, ntrain, nwarmup=defaults['nwarmup'], batch_size=defaults['batch_size'],
+                 convergence_criterion=defaults['convergence_criterion'], nsearch=defaults['nsearch'],
+                 make_plot=defaults['make_plot']):
         self.runner = runner
-        self.surrogate = surrogate
         self.variables = variables
 
-        self.search_space = {var.name: np.linspace(*var.constraints, nsearch_points)
-                             for var in variables.list if var.kind.lower() in 'activelearning'}
-
-        al_keys = [var.name for var in variables.list if var.kind.lower() == 'activelearning']
-        Xpred = [np.linspace(*var.constraints, nsearch_points) if var.name in al_keys else np.unique(var.value)
-                 for var in variables.input_list]
-        self.Xpred = np.hstack([xi.flatten().reshape(-1, 1) for xi in np.meshgrid(*Xpred)])
-
-        self.nwarm = nwarm
         self.ntrain = ntrain
+        self.nwarmup = nwarmup
+
         self.batch_size = batch_size
+        if (ntrain - nwarmup) % batch_size:
+            raise RuntimeError("Number of learning points ({}) and batch size ({}) for AL not compatible!"
+                               .format(ntrain - nwarmup, batch_size))
```

Comment:
We should probably discuss a coding style at some point to decide whether we prefer f-strings or `.format`. Usually consistency is desirable.